programs/stake/src/stake_state.rs

```diff
@@ -766,13 +770,31 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
                         return Err(InstructionError::InsufficientFunds);
                     }
                     // split the stake, subtract rent_exempt_balance unless
-                    //  the destination account already has those lamports
-                    //  in place.
-                    // this could represent a small loss of staked lamports
-                    //  if the split account starts out with a zero balance
-                    let split_stake = stake.split(
-                        lamports - meta.rent_exempt_reserve.saturating_sub(split.lamports()?),
-                    )?;
+                    // the destination account already has those lamports
+                    // in place.
+                    // this means that the new stake account will have a stake equivalent to
+                    // lamports minus rent_exempt_reserve if it starts out with a zero balance
+                    let (remaining_stake_delta, split_stake_amount) = if lamports
+                        == self.lamports()?
+                    {
+                        // If split amount equals the full sounce stake, the new split stake must
+                        // equal the same amount, regardless of any current lamport balance in the
+                        // split account. Since split accounts retain the state of their sournce
```

Comment:
```suggestion
                        // split account. Since split accounts retain the state of their source
```

Reply:
🤦‍♀️ How many wrong ways can I spell that word?

Reply:
I believe company policy is to blame the Macbook Pro keyboard!